xknx/config/config.py

```diff
@@ -35,16 +35,17 @@ def read(self, file: str = "xknx.yaml") -> None:
         """Read config."""
         logger.debug("Reading %s", file)
         doc = load_yaml(file)
+        assert isinstance(doc, dict)
```

Comment:
I think this is safe to assert here. 
`load_yaml` returns `JSON_TYPE = Union[List, Dict, str]` but as far as I understood the code it should always return `OrderedDict, Dict` - the config would be invalid if it didn't too...

Reply:
We could also just `cast` it and call it a day...